File: src/main/java/com/harleylizard/magic_things/common/MagicThingsBiomes.java
package com.harleylizard.magic_things.common;

import net.minecraft.core.registries.Registries;
import net.minecraft.resources.ResourceKey;
import net.minecraft.world.level.biome.Biome;

public final class MagicThingsBiomes {
    public static final ResourceKey<Biome> FOULED_FOREST = biomeOf("fouled_forest");
    public static final ResourceKey<Biome> FOULED_PLAINS = biomeOf("fouled_plains");

    public static ResourceKey<Biome> biomeOf(String name) {
        return ResourceKey.create(Registries.BIOME, MagicThings.resourceLocation(name));
    }

}

File: src/main/kotlin/com/harleylizard/magic_things/common/terrablender/MagicThingsBiomes.kt
package com.harleylizard.magic_things.common.terrablender

import com.harleylizard.magic_things.common.MagicThings.Companion.resourceLocation
import net.minecraft.core.registries.Registries
import net.minecraft.resources.ResourceKey
import net.minecraft.world.level.biome.Biome

object MagicThingsBiomes {
    val fouledForest = "fouled_forest".resourceKey
    val fouledPlains = "fouled_plains".resourceKey

    val String.resourceKey: ResourceKey<Biome> get() = ResourceKey.create(Registries.BIOME, resourceLocation)

}